include/zephyr/drivers/i2c.h

```diff
@@ -599,6 +607,63 @@ static inline int z_impl_i2c_transfer(const struct device *dev,
 	return res;
 }
 
+
+/**
+ * @brief Perform data transfer to another I2C device in master mode.
+ *
+ * This routine provides a generic interface to perform data transfer
+ * to another I2C device asynchronously.
+ *
+ * The array of message @a msgs must not be NULL.  The number of
+ * message @a num_msgs may be zero,in which case no transfer occurs.
+ *
+ * @note Not all scatter/gather transactions can be supported by all
+ * drivers.  As an example, a gather write (multiple consecutive
+ * `i2c_msg` buffers all configured for `I2C_MSG_WRITE`) may be packed
+ * into a single transaction by some drivers, but others may emit each
+ * fragment as a distinct write transaction, which will not produce
+ * the same behavior.  See the documentation of `struct i2c_msg` for
+ * limitations on support for multi-message bus transactions.
+ *
+ * @param dev Pointer to the device structure for an I2C controller
+ * driver configured in master mode.
+ * @param msgs Array of messages to transfer.
+ * @param num_msgs Number of messages to transfer.
+ * @param addr Address of the I2C target device.
+ * @param callback Function pointer for completion callback. See `async.h` for
```

Comment:
`@param callback` -> `@param cb` to align with prototype and avoid doc build failure in CI